Tests/SonarScanner.MSBuild.PreProcessor.Test/Infrastructure/MockObjectFactory.cs

```diff
@@ -61,8 +61,10 @@ public MockObjectFactory(bool withDefaultRules = true, string organization = nul
         public ITargetsInstaller CreateTargetInstaller() =>
             TargetsInstaller.Object;
 
-        public IAnalyzerProvider CreateRoslynAnalyzerProvider(ISonarWebServer server) =>
-            AnalyzerProvider;
+        public IAnalyzerProvider CreateRoslynAnalyzerProvider(ISonarWebServer server, string localCacheTempPath)
+        {
+            return AnalyzerProvider = new(new MockAnalyzerInstaller(localCacheTempPath)) { SettingsToReturn = new AnalyzerSettings { RulesetPath = "c:\\xxx.ruleset" } };
```

Comment:
The created installer is not actually needed, what is needed is the folder the constructor creates.